spec/compiler/macro/macro_methods_spec.cr

```diff
@@ -2546,6 +2546,21 @@ module Crystal
       end
     end
 
+    describe MacroExpression do
+      foo = MacroExpression.new(NilLiteral.new, output: false)
+      bar = MacroExpression.new(1.int32, output: true)
+
+      it "executes exp" do
+        assert_macro %({{x.exp}}), "nil", {x: foo}
```

Comment:
**suggestion:** I find these expectations unnecessarily hard to read when the definition of the input parameters is moved out of context. In order to understand this expectation, I need to look at the definition of `foo` a couple lines above. It could easily be inlined, which IMO improves readability a lot:
```suggestion
        assert_macro %({{x.exp}}), "nil", {x: MacroExpression.new(NilLiteral.new, output: false)}
```
(is `output: false` even necessary for this spec?)

I don't think there's any relevant gain from avoiding duplication here. I'd certainly prefer explicitness.